File: pmdash_ui/src/models/task.ts
import { Type } from '../enums/type';
import { Status } from '../enums/status';
import { Priority } from '../enums/priority';

export default class Task {
  title = '';
  type = Type.task;
  priority = Priority.low;
  lables = '';
  sprint = '';
  status = Status.backlog;
  discription = '';
  attachments: string[] = [];
  subTaskes: string[] = [];
  assigned = '';
  created = new Date();
  subTasks: Task[] = [];
}

File: pmdash_api/models/task.py
from typing import List
from datetime import datetime
from bson import ObjectId
from pydantic import BaseModel, Field
from utilities.utils import PyObjectId


class Task(BaseModel):
    id: PyObjectId = Field(default_factory=PyObjectId, alias="_id")
    title: str = Field(...)
    type: str = Field(...)
    priority: str = Field(...)
    lables: str = Field(...)
    sprint: str = Field(...)
    status: str = Field(...)
    discription: str = Field(...)
    attachments: List[str] = Field(...)
    subTaskes: List[str] = Field(...)
    assigned: str = Field(...)
    created: datetime = Field(...)
    subTasks: List['Task'] = Field(...)

    class Config:
        allow_population_by_field_name = True
        arbitrary_types_allowed = True
        json_encoders = {ObjectId: str}
